### ISSUE TITLE
Demo is not working

### ISSUE THREAD

**Issue by faouzioudouh** [valor-software/ng2-table#384]
Please, how i can get Demo to work

Thanks

**Comment by webcat12345** [comment]
How did you run the demo?

**Comment by faouzioudouh** [comment]
I've tried to build the demo with `webpack` and then add files to `index.html` inside demo folder.
what do you reckon?

**Comment by webcat12345** [comment]
Which error you'v got?

**Comment by faouzioudouh** [comment]
Thank you for your help in advance.
![github](https://cloud.githubusercontent.com/assets/4366612/21639818/0d625a7a-d26a-11e6-8929-8aeb27842851.PNG)

**Comment by webcat12345** [comment]
I see, I think you didn't installed modules and packages correctly. However it works well with others but you 'v got error so we need to fix it. 
Please check this : https://github.com/angular/angular/issues/7717

**Comment by faouzioudouh** [comment]
Yes, running `npm install` throws bunch of errors including some for missing a python on my machine, why python is required here?

**Comment by webcat12345** [comment]
That's it. You need to complete installing module without any error.
Regarding to pythons there needs to ruby and others also.

**Comment by faouzioudouh** [comment]
OK. assuming i've done that without errors, Which commande will build the demo? thought i didn't understand the `webpack.config.js` included

**Comment by webcat12345** [comment]
It is depends which frameworks you used. mostly I used `angualr-cli`  : `ng serve`

**Comment by webcat12345** [comment]
Also other boiler plate code you use for example scotch-ng-router. in this case you can use `npm start`.

**Comment by faouzioudouh** [comment]
Actually i don't know what they used to build this project, i'll try
`angular-cli` and let you know if it works.
Thank you so much sir.

*Faouzi <https://ma.linkedin.com/in/foudouh>*

**Comment by webcat12345** [comment]
You are welcome.

**Comment by protagonist51** [comment]
@webcat12345 Now it has 3 sorting options, ASC, DESC, and nothing(?).
But I'd like to use ASC and DESC only, is it possible I can set them?
Please advise me.

**Comment by webcat12345** [comment]
I didn't find out option for that yet. But I think it is possible to do that.

**Comment by protagonist51** [comment]
@webcat12345 Please advice me. I'm really looking for that solution. And is it possible to change background color of heading columns?

**Comment by webcat12345** [comment]
You can customize component as below.

**ng2-table/components/table/ng-table-sorting.directive.js   line:17**

Then you can see these code 

```
    NgTableSortingDirective.prototype.onToggleSort = function (event) {
        if (event) {
            event.preventDefault();
        }
        if (this.ngTableSorting && this.column && this.column.sort !== false) {
            switch (this.column.sort) {
                case 'asc':
                    this.column.sort = 'desc';
                    break;
                case 'desc':
                    this.column.sort = '';
                    break;
                default:
                    this.column.sort = 'asc';
                    break;
            }
            this.sortChanged.emit(this.column);
        }
    };

```

This is code for switching sorting mode. 
If you change this as below then you can use only two methods.

```
    NgTableSortingDirective.prototype.onToggleSort = function (event) {
        if (event) {
            event.preventDefault();
        }
        if (this.ngTableSorting && this.column && this.column.sort !== false) {
            switch (this.column.sort) {
                case 'asc':
                    this.column.sort = 'desc';
                    break;
                case 'desc':
                    this.column.sort = 'asc';
                    break;
                default:
                    this.column.sort = 'asc';
                    break;
            }
            this.sortChanged.emit(this.column);
        }
    };
```

You can fort repository and use component as freely :)
